apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/site-description/edit.js

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { PlainText } from '@wordpress/editor';
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+
+class SiteDescriptionEdit extends Component {
+	state = {
+		fromApi: null,
+		value: __( 'Site description loading…' ),
+	};
+
+	componentDidMount() {
+		try {
+			apiFetch( { path: '/wp/v2/settings' } ).then( ( { description } ) => {
+				this.setState( {
+					fromApi: description,
+					value: description,
+				} );
+			} );
+		} catch ( error ) {
+			this.handleApiError();
+		}
+	}
+
+	componentDidUpdate( prevProps ) {
+		if (
+			prevProps.isSaving === false &&
+			this.props.isSaving === true &&
```

Comment:
This is also updating the site description during autosaves (i.e. when clicking on the Preview button), but I think we should only update it when the user explicitly has clicked on the Publish/Update button.